src/storage/engine/mod.rs

```diff
@@ -381,8 +393,22 @@ mod tests {
         assert_near_seek(cursor_mut, b"y", (b"z", b"2"));
         assert_near_seek(cursor_mut, b"x\x00", (b"z", b"2"));
         assert!(!cursor_mut.near_seek(&make_key(b"z\x00")).unwrap());
+        for i in 0..100 {
+            let key = format!("y{}", i);
+            must_put(engine, &key.as_bytes(), b"3");
+        }
+        let snapshot = engine.snapshot(&Context::new()).unwrap();
+        let mut cursor = snapshot.iter(None).unwrap();
+        let cursor_mut = cursor.as_mut();
+        assert_near_seek(cursor_mut, b"x", (b"x", b"1"));
+        assert_near_seek(cursor_mut, b"z", (b"z", b"2"));
+
         must_delete(engine, b"x");
         must_delete(engine, b"z");
+        for i in 0..100 {
+            let key = format!("y{}", i);
+            must_delete(engine, &key.as_bytes());
```

Comment:
Insert many key-values between 'x' and 'z' then`near_seek` will fallback to `seek`.